### ISSUE TITLE
Program in C: Linux Development Basics

### ISSUE THREAD

**Issue by hpdang** [fossasia/Codeheat#10]
This issue can be taken by multiple CodeHeat participants. Anyone can grab this issue at any time. Once you are done, submit the outcome via the form below.

### OBJECTIVE
Learn about basic development techniques on the Linux platform by learning how to develop a simple C language program with vim, how to build it with GCC and run it.

### REQUIREMENTS
- Follow [the openEuler mirror repository on GitHub](https://github.com/openeuler-mirror) indicating your participation 
- Learn basic tools and commands that could be used while software development on Linux platform
- Follow [this instruction](https://github.com/openeuler-mirror/openEulerCodeHeat/blob/main/Advanced-Issues/Linux-Dev-Basics.md) 
- Complete all the steps 
- Take a screenshot of the output at the final step (a calculation of cos and sin value) 

### EXPECTED OUTCOME
- Learn how to use basic tools and command lines 
- Successfully build and run a simple program on Linux platform

### HOW TO SUBMIT YOUR WORK
- Submit your screenshot [here](https://docs.google.com/forms/d/e/1FAIpQLSfPyHBA_thhNzs7pBVj7tBsiml_ZkY0PBZPpN_fnd3r4UxWhQ/viewform) 

### QUESTIONS
- If you have any questions, please comment on this issue directly, or reach out to us at https://gitter.im/fossasia/fossasia
- If you identify any problem while working on this issue, take up [a related task](https://github.com/fossasia/Codeheat/issues/13) to file a bug report

**Comment by MattXLei** [comment]
Hi, when working on this issue, I got a few warnings and errors when compiling the 4 C files at the end of Step 3, as shown in the picture below. I fixed them on my end and can share my code if needed.

<img width="700" alt="Errors" src="https://github.com/fossasia/Codeheat/assets/49255967/f3121e55-b0aa-4579-bc70-56b42ff8bf19">

**Comment by Labnann** [comment]
> Hi, when working on this issue, I got a few warnings and errors when compiling the 4 C files at the end of Step 3, as shown in the picture below. I fixed them on my end and can share my code if needed.
> <img alt="Errors" width="700" src="https://private-user-images.githubusercontent.com/49255967/291948954-f3121e55-b0aa-4579-bc70-56b42ff8bf19.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwOTE0NzcsIm5iZiI6MTcwMzA5MTE3NywicGF0aCI6Ii80OTI1NTk2Ny8yOTE5NDg5NTQtZjMxMjFlNTUtYjBhYS00NTc5LWJjNzAtNTZiNDJmZjhiZjE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjIwVDE2NTI1N1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTEwODVlOThjM2ViYmQxNTBkNjM2MzJjMmMxNGVjMDFlOTk5ZmQ2N2RiNTYyYzhmZWRlY2NhMmUzY2RmODMzN2UmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.6FCflwiaYaoYv_BE9jI3vNKo6MNcyZQWpzYXIs7VE9o">

I think this issue is fixed now. This is the code on my end :).

![Screenshot_20231220_225501](https://github.com/fossasia/Codeheat/assets/55809005/913adc79-4641-46f3-b144-1fa183f8ea0a)

Even so, I don't think the function cos_value should be header guarded with 
SIN_VALUE

**Comment by MattXLei** [comment]
Yeah, one of the changes I made was replacing all the SIN_VALUE_H in cos_value.h with COS_VALUE_H, and the code worked as a result.

**Comment by ankur0904** [comment]
@hpdang Do we need to only submit the form or need to make the PR?

**Comment by hpdang** [comment]
@ankur0904 only submit the form, thanks!

**Comment by ankur0904** [comment]
@hpdang Thanks!

**Comment by ankur0904** [comment]
@hpdang 

The code provided in the zip file has the following files-
1. cos_value.c
2. cos_value.h
3. hello.c
4. main.c
5. sin_value.c
6. sin_value.h

And the `main.c` code looks like that

```C
#include <stdio.h>
#include "sin_value.h"
#include "cos_value.h"

#define pi 3.14159

char name[255];
float angle;

int main(void)
{
    printf("\n\nPlease input your name: ");
    scanf("%s", &name);
    printf("\nPlease enter the degree angle (ex> 90): ");
    scanf("%f", &angle);
    printf("\nInput angle: %5.2f", angle);
    hello(name);           //// <---------------- ////
    sin_value(angle);
    cos_value(angle);
    return 0;
}

```
But in the `main.c`, we are calling the `hello` function with the **name** argument, but we have not included the **name** function like `sin_value, cos_value` in `main.c` file so when we compile these files we get the warning like this

```bash
main.c:17:5: warning: implicit declaration of function ‘hello’; did you mean ‘ftello’? [-Wimplicit-function-declaration]
   17 |     hello(name);
      |     ^~~~~
      |     ftello
```

**Solution to this problem:**
Create a new file `hello.h` with the following code
```C
#ifndef HELLO_H
#define HELLO_H

void hello(char* name);

#endif
```
And in `main.c`
```C
#include <stdio.h>
#include "sin_value.h"
#include "cos_value.h"
#include "hello.h"        // <---------- Add the code // 

#define pi 3.14159

...

```

After that, the code is compiled and consistent with the remaining files `sin_value.c, cos_value.c`.

**Comment by ankur0904** [comment]
If the issue is valid I am happy to submit a PR

**Comment by hpdang** [comment]
@ankur0904 yes, please feel free to submit a PR and you can take up [another task ](https://github.com/fossasia/Codeheat/issues/13) to file a report.

**Comment by ankur0904** [comment]
@hpdang I have noticed that the C program files are zipped under the `tar` file [here](https://github.com/openeuler-mirror/openEulerCodeHeat/blob/main/Advanced-Issues/awsome_codes.tar.gz). So I am confused how to make changes in the file. Meanwhile, I will look at [another task](https://github.com/fossasia/Codeheat/issues/13)

Thanks

**Comment by Labnann** [comment]
> @hpdang I have noticed that the C program files are zipped under the `tar` file [here](https://github.com/openeuler-mirror/openEulerCodeHeat/blob/main/Advanced-Issues/awsome_codes.tar.gz). So I am confused how to make changes in the file. Meanwhile, I will look at [another task](https://github.com/fossasia/Codeheat/issues/13)
> 
> Thanks

Opening a issue there may be a good idea.

**Comment by hongquan** [comment]
Perhaps the bug in *awsome_codes.tar.gz* is an exercise for you to do [Identify an issue and file a bug report on openEuler](https://github.com/fossasia/Codeheat/issues/13)?

**Comment by varanasiroshan2001** [comment]
Done this issue succesfully, no issues ran in while i was doing it in AWS Redhat based Linux VM 

![image](https://github.com/fossasia/Codeheat/assets/64471180/94e955fb-77dd-4763-a6d2-eb2d4fd62114)
![image](https://github.com/fossasia/Codeheat/assets/64471180/8e6ab28a-ec28-4eec-a83d-1acd9a0cd0d1)

**Comment by Satarupa22-SD** [comment]
hi, I am unable to run the tar files on macbook, can anyone please help . Thanks!
![Uploading Screenshot 2024-01-08 at 12.26.04 AM.png…]()

**Comment by Satarupa22-SD** [comment]
> Hi, when working on this issue, I got a few warnings and errors when compiling the 4 C files at the end of Step 3, as shown in the picture below. I fixed them on my end and can share my code if needed.
> 
> <img alt="Errors" width="700" src="https://private-user-images.githubusercontent.com/49255967/291948954-f3121e55-b0aa-4579-bc70-56b42ff8bf19.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDQ2NTQ0NDUsIm5iZiI6MTcwNDY1NDE0NSwicGF0aCI6Ii80OTI1NTk2Ny8yOTE5NDg5NTQtZjMxMjFlNTUtYjBhYS00NTc5LWJjNzAtNTZiNDJmZjhiZjE5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAxMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMTA3VDE5MDIyNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTkzZDY3ZTcyNTE1ODQ2OTA1MTRkZmRmYWEzZmRkNTRmMDRiMzI5MTNjMDY0NzI2YWFiZWQ2ZmMzMmQyNDdiYWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.9sFpdTJMKAssCHnipaJb0ZBfoTQJm20MMVPppOXWAaI">

hi, i got the following issues while working, can you please help
curl: (6) Could not resolve host: s
Warning: Binary output can mess up your terminal. Use "--output -" to tell 
Warning: curl to output it to your terminal anyway, or consider "--output 
Warning: <FILE>" to save to a file.

**Comment by Pdahiya5033** [comment]
hey @hpdang !, can i work on issue. I am eager to learn linux
